Sources/NIOConcurrencyHelpers/fastatomics.swift

```diff
@@ -0,0 +1,271 @@
+    public typealias AtomicWrapper = catmc_fast_atomic_unsigned_long_long
+    public static let fast_atomic_create_with_existing_storage = catmc_fast_atomic_unsigned_long_long_create_with_existing_storage
+    public static let fast_atomic_compare_and_exchange         = catmc_fast_atomic_unsigned_long_long_compare_and_exchange
+    public static let fast_atomic_add                          = catmc_fast_atomic_unsigned_long_long_add
+    public static let fast_atomic_sub                          = catmc_fast_atomic_unsigned_long_long_sub
+    public static let fast_atomic_exchange                     = catmc_fast_atomic_unsigned_long_long_exchange
+    public static let fast_atomic_load                         = catmc_fast_atomic_unsigned_long_long_load
+    public static let fast_atomic_store                        = catmc_fast_atomic_unsigned_long_long_store
+}
+
+extension Int: FastAtomicPrimitive {
+    public typealias AtomicWrapper = catmc_fast_atomic_long
+    public static let fast_atomic_create_with_existing_storage = catmc_fast_atomic_long_create_with_existing_storage
+    public static let fast_atomic_compare_and_exchange         = catmc_fast_atomic_long_compare_and_exchange
+    public static let fast_atomic_add                          = catmc_fast_atomic_long_add
+    public static let fast_atomic_sub                          = catmc_fast_atomic_long_sub
+    public static let fast_atomic_exchange                     = catmc_fast_atomic_long_exchange
+    public static let fast_atomic_load                         = catmc_fast_atomic_long_load
+    public static let fast_atomic_store                        = catmc_fast_atomic_long_store
+}
+
+extension UInt: FastAtomicPrimitive {
+    public typealias AtomicWrapper = catmc_fast_atomic_unsigned_long
+    public static let fast_atomic_create_with_existing_storage = catmc_fast_atomic_unsigned_long_create_with_existing_storage
+    public static let fast_atomic_compare_and_exchange         = catmc_fast_atomic_unsigned_long_compare_and_exchange
+    public static let fast_atomic_add                          = catmc_fast_atomic_unsigned_long_add
+    public static let fast_atomic_sub                          = catmc_fast_atomic_unsigned_long_sub
+    public static let fast_atomic_exchange                     = catmc_fast_atomic_unsigned_long_exchange
+    public static let fast_atomic_load                         = catmc_fast_atomic_unsigned_long_load
+    public static let fast_atomic_store                        = catmc_fast_atomic_unsigned_long_store
+}
+
+public class FastAtomic<T: FastAtomicPrimitive> {
+    @usableFromInline
+    typealias Manager = ManagedBufferPointer<Void, T.AtomicWrapper>
+
+    /// Create an atomic object with `value`
+    @inlinable
+    static func makeAtomic(value: T) -> FastAtomic {
+        let manager = Manager(bufferClass: self, minimumCapacity: 1) { _, _ in }
+        let fastAtomic = unsafeDowncast(manager.buffer, to: FastAtomic<T>.self)
```

Comment:
I don't love `unsafeDowncast` here: is there any reason not to have the type checker validate we got this right with `as!`?

Reply:
```as!``` should be fine. I'll change it.